### PR TITLE
feat: single annotation dialog step 2

### PR DESCRIPTION
**Pull request by codemonkey800**
#450

Implements step 2 for the download annotation dialog

- Refactor to pass `activeAnnotation` instead of annotation map
- Adds `react-query` for client side fetching
- Refactor file size fetching to be more efficient
  - instead of fetching all file sizes for the page, fetch on demand when user views file size in the dialog
- Fix navigation for step 2

## Demos

https://dev-jeremy-dl-modal.cryoet.dev.si.czi.technology

<details>
<summary>Expand to see demos</summary>

### Configure Download

image

#### `.mrc` Direct Download

image

#### `.mrc` Curl

image

#### `.mrc` AWS

image

#### `.mrc` API

image

#### `.ndjson` Direct Download

image

#### `.ndjson` Curl

image

#### `.ndjson` AWS

image

#### `.ndjson` API

image

#### `.zarr` AWS

image

#### `.zarr` API

image

</details>

### CODE REVIEW

File: frontend/packages/data-portal/app/components/Download/APIDownloadTab.tsx [hunk 1]
```diff
@@ -14,46 +14,83 @@ export function APIDownloadTab() {
...
         )
+        .with({ type: 'annotation' }, () => ({
+          label: t('copyApiCodeSnippet'),
+          content: (
+            <>
+              from cryoet_data_portal import Client, Annotation
+              <br />
+              client = Client()
+              <br />
+              annotation = next(Annotation.find(client, [id == {annotationId}]))
```
Comment (manasaV3) [inline frontend/packages/data-portal/app/components/Download/APIDownloadTab.tsx:51]: I believe this should be 
```
from cryoet_data_portal import Client, Annotation
client = Client()
annotation = Annotation.find(client, [id == {annotationId}])
annotation.download()
```

File: frontend/packages/data-portal/app/components/Download/ConfigureAnnotationDownloadContent.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+import { FileFormatDropdown } from './FileFormatDropdown'
+
+export function ConfigureAnnotationDownloadContent() {
+  const { objectShapeType } = useDownloadModalQueryParamState()
+  const { activeAnnotation } = useDownloadModalContext()
+
+  const fileFormats = useMemo<string[]>(
+    () =>
+      activeAnnotation?.files
```
Comment (kne42) [inline frontend/packages/data-portal/app/components/Download/ConfigureAnnotationDownloadContent.tsx:14]: oops yeah this is way more efficient

File: frontend/packages/data-portal/app/components/Download/ConfigureTomogramDownloadContent.tsx [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+import { useDownloadModalQueryParamState } from 'app/hooks/useDownloadModalQueryParamState'
+import { useI18n } from 'app/hooks/useI18n'
+import { useLogPlausibleCopyEvent } from 'app/hooks/useLogPlausibleCopyEvent'
+import { DownloadConfig } from 'app/types/download'
+import { cns } from 'app/utils/cns'
+
+import { CopyBox } from '../CopyBox'
+import { Select, SelectOption } from '../Select'
+
+function Radio({
```
Comment (kne42) [inline frontend/packages/data-portal/app/components/Download/ConfigureTomogramDownloadContent.tsx:23]: could we maybe extract this to another file?

File: frontend/packages/data-portal/app/routes/api.file-size.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  if (fileUrl) {
+    const res = await axios.head(fileUrl)
+    const contentLength = res.headers['content-length'] as string | null
+
+    if (contentLength) {
+      fileSize = isNumber(+contentLength) ? +contentLength : undefined
+    }
+  }
+
+  return json({ fileSize })
```
Comment (kne42) [inline frontend/packages/data-portal/app/routes/api.file-size.ts:27]: nice

**Review by kne42 — APPROVED** [review]
lgtm code-wise! love the refactors!

**Comment by kev-zunshiwang** [comment]
Hey Jeremy! Looking good!

I have the feedback in the google doc below, can you cross out the ones that are fixed and you can leave a comment for the ones you have questions.

Thank you so much!!

Feedback doc: https://docs.google.com/document/d/1QyIex7NhRj53a2BOzps-HWJAPpXc6UDkyPX8XnQ8M-U/edit#heading=h.wc4bnytgbsc0

**Comment by Janeece** [comment]
@codemonkey800 I noticed that the subhead copy inside all of the AWS CLI tabs still says "AWS S3" -- could you update that as well?

**Pull request merged**